kotlintest-assertions/src/commonMain/kotlin/io/kotlintest/matchers/collections/matchers.kt

```diff
@@ -491,6 +475,83 @@ fun <T> beOneOf(collection: Collection<T>) = object : Matcher<T> {
   }
 }
 
+/**
+ * Verifies that this element is in [collection] by comparing value
+ *
+ * Assertion to check that this element is in [collection]. This assertion checks by value, and not by reference,
+ * therefore even if the exact instance is not in [collection] but another instance with same value is present, the
+ * test will pass.
+ *
+ * An empty collection will always fail. If you need to check for empty collection, use [Collection.shouldBeEmpty]
+ *
+ * @see [shouldNotBeIn]
+ * @see [beIn]
+ */
+infix fun <T> T.shouldBeIn(collection: Collection<T>) = this should beIn(collection)
+
+/**
+ * Verifies that this element is NOT any of [collection]
+ *
+ * Assertion to check that this element is not any of [collection]. This assertion checks by value, and not by reference,
+ * therefore any instance with same value must not be in [collection], or this will fail.
+ *
+ * An empty collection will always fail. If you need to check for empty collection, use [Collection.shouldBeEmpty]
+ *
+ * @see [shouldNotBeIn]
+ * @see [beIn]
+ */
+infix fun <T> T.shouldNotBeIn(collection: Collection<T>) = this shouldNot beIn(collection.toList())
+
+/**
+ * Verifies that this element is any of [any] by comparing value
+ *
+ * Assertion to check that this element is any of [any]. This assertion checks by value, and not by reference,
+ * therefore even if the exact instance is not any of [any] but another instance with same value is present, the
+ * test will pass.
+ *
+ * An empty collection will always fail. If you need to check for empty collection, use [Collection.shouldBeEmpty]
+ *
+ * @see [shouldNotBeIn]
+ * @see [beIn]
+ */
+fun <T> T.shouldBeIn(vararg any: T) = this should beIn(any.toList())
+
+/**
+ * Verifies that this element is NOT any of [any]
+ *
+ * Assertion to check that this element is not any of [any]. This assertion checks by value, and not by reference,
+ * therefore any instance with same value must not be in [any], or this will fail.
+ *
+ * An empty collection will always fail. If you need to check for empty collection, use [Collection.shouldBeEmpty]
+ *
+ * @see [shouldNotBeIn]
+ * @see [beIn]
+ */
+fun <T> T.shouldNotBeIn(vararg any: T) = this shouldNot beIn(any.toList())
+
+/**
+ *  Matcher that verifies that this element is in [collection] by comparing value
+ *
+ * Assertion to check that this element is in [collection]. This assertion checks by value, and not by reference,
+ * therefore even if the exact instance is not in [collection] but another instance with same value is present, the
+ * test will pass.
+ *
+ * An empty collection will always fail. If you need to check for empty collection, use [Collection.shouldBeEmpty]
+ *
+ * @see [shouldBeOneOf]
+ * @see [shouldNotBeOneOf]
+ */
+fun <T> beIn(collection: Collection<T>) = object : Matcher<T> {
+  override fun test(value: T): MatcherResult {
+    if (collection.isEmpty()) throwEmptyCollectionError()
+
+    val match = collection.any { it == value }
```

Comment:
Wouldn't a `it in collection` work? This will probably optimize collections based on `hashcode`